unblob/handlers/filesystem/squashfs.py

```diff
@@ -27,7 +28,12 @@ def calculate_chunk(self, file: File, start_offset: int) -> Optional[ValidChunk]
         header = self.parse_header(file, endian)
 
         end_of_data_offset = start_offset + header.bytes_used
-        file.seek(end_of_data_offset)
+        try:
+            file.seek(end_of_data_offset, io.SEEK_SET)
+        except ValueError:
+            raise InvalidInputFormat(
+                "Squashfs bytes_used value is larger than file size."
+            ) from ValueError
```

Comment:
I do not really understand this. The File seek already throws either a SeekError or a ValueError (in case we also read an EOFError).

Which is handled in finder.py:
```py
def _calculate_chunk(
    handler: Handler, file: File, real_offset, task_result: TaskResult
) -> Optional[ValidChunk]:
    file.seek(real_offset)
    try:
        return handler.calculate_chunk(file, real_offset)
    except InvalidInputFormat as exc:
        logger.debug(
            "File format is invalid",
            exc_info=exc,
            handler=handler.NAME,
            _verbosity=2,
        )
    except EOFError as exc:
        logger.debug(
            "File ends before header could be read",
            exc_info=exc,
            handler=handler.NAME,
            _verbosity=2,
        )
    except SeekError as exc:
        logger.debug(
            "Seek outside file during chunk calculation",
            exc_info=exc,
            handler=handler.NAME,
            _verbosity=2,
        )
```

Why is it better to have an InvalidFormat instead of any of the above?

Reply:
Removed the squashfs commit and kept the YAFFS one.